harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -99,6 +139,406 @@ def test_minimal_vm(api_client, image_id, unique_vm_name, wait_timeout):
+                                break
+                            sleep(3)
+                        else:
+                            raise AssertionError(
+                                f"Failed to Delete VM({unique_vm_name}) with errors:\n"
+                                f"Status({code}): {data}"
+                            )
+
+                        fails, check = [], dict()
+                        for vol in spec.volumes:
+                            vol_name = vol['volume']['persistentVolumeClaim']['claimName']
+                            check[vol_name] = api_client.volumes.delete(vol_name)
+
+                        while endtime > datetime.now():
+                            l_check = dict()
+                            for vol_name, (code, data) in check.items():
+                                if 200 != code:
+                                    fails.append(
+                                        (vol_name, f"Failed to delete\nStatus({code}): {data}"))
+                                else:
+                                    code, data = api_client.volumes.get(vol_name)
+                                    if 404 != code:
+                                        l_check[vol_name] = (code, data)
+                            check = l_check
+                            if not check:
+                                break
+                            sleep(5)
+                        else:
+                            for vol_name, (code, data) in check.items():
+                                fails.append(
+                                    (vol_name, f"Failed to delete\nStatus({code}): {data}"))
+
+                        assert not fails, (
+                            f"Failed to delete VM({unique_vm_name})'s volumes with errors:\n"
+                            "\n".join(f"Volume({n}): {r}" for n, r in fails)
+                        )
+                        break
+                    sleep(3)
+                else:
+                    raise AssertionError(
+                        f"Failed to Start VM({unique_vm_name}) with errors:\n"
+                        f"Status({code}): {data}"
+                    )
+            else:
+                raise AssertionError(
+                    f"Failed to Update VM({unique_vm_name}) with errors:\n"
+                    f"Phase: {data.get('status', {}).get('phase')}\t"
+                    f"Status: {data.get('status')}\n"
+                    f"API Status({code}): {data}"
+                )
+            break
+        sleep(3)
+    else:
+        raise AssertionError(
+            f"Failed to create VM({cpu} core, {mem} RAM) with errors:\n"
+            f"Phase: {data.get('status', {}).get('phase','')}\t"
+            f"Status: {data.get('status', {})}\n"
+            f"API Status({code}): {data}"
+        )
+
+
+@pytest.mark.p0
+@pytest.mark.virtualmachines
+def test_vm_with_bogus_vlan(api_client, image_id, unique_vm_name, wait_timeout, bogus_vlan_net):
```

Comment:
this should be marked as negative test

Reply:
I did forget that marker, great catch smile - updated +1